src/main/java/picard/util/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,260 @@
+            if (ctx.getAlleles().size() == 2) return ctx;
+
+            return AlleleSubsettingUtils.subsetAlleles(ctx, Arrays.asList(ctx.getReference(), altAlleleMaybe.get()));
+        }
+
+        // if not, perhaps we have NON_REF_ALLELE
+        final Optional<Allele> nonRefAlleleMaybe = ctx.getAlternateAlleles().stream().filter(a -> a.equals(NON_REF_ALLELE)).findAny();
+        if (nonRefAlleleMaybe.isPresent()) {
+
+            final VariantContext vcSubsetted = ctx.getAlleles().size() == 2 ? ctx : AlleleSubsettingUtils.subsetAlleles(ctx, Arrays.asList(ctx.getReference(), nonRefAlleleMaybe.get()));
+            return AlleleSubsettingUtils.swapAlleles(vcSubsetted, NON_REF_ALLELE, Allele.create(otherAllele));
+        }
+        return null;
+    }
+
+    public static VariantContext subsetAlleles(final VariantContext originalVc, final List<Allele> allelesToKeep) {
+        VariantContextBuilder vcBuilder = new VariantContextBuilder(originalVc).alleles(allelesToKeep);
+
+        GenotypesContext newGenotypes = subsetAlleles(originalVc.getGenotypes(), originalVc.getAlleles(), allelesToKeep);
+        vcBuilder.genotypes(newGenotypes);
+        return vcBuilder.make();
+    }
+
+    /**
+     * swaps one of the alleles in a VC (and its genotypes) with another.
+     */
+    public static VariantContext swapAlleles(final VariantContext originalVc, final Allele oldAllele, final Allele newAllele) {
+        if (!originalVc.getAlleles().contains(oldAllele)) return originalVc;
+
+        final LinkedList<Allele> alleles = new LinkedList<>(originalVc.getAlleles());
+        alleles.set(alleles.indexOf(oldAllele), newAllele);
+
+        VariantContextBuilder vcBuilder = new VariantContextBuilder(originalVc).alleles(alleles);
+        GenotypesContext newGTs = GenotypesContext.create(originalVc.getGenotypes().size());
+
+        for (final Genotype g : originalVc.getGenotypes()) {
+            if (!g.getAlleles().contains(oldAllele)) {
+                newGTs.add(g);
+            } else {
+                final GenotypeBuilder gb = new GenotypeBuilder(g);
+                final LinkedList<Allele> genotypeAlleles = new LinkedList<>(g.getAlleles());
+                genotypeAlleles.set(genotypeAlleles.indexOf(oldAllele), newAllele);
+                gb.alleles(genotypeAlleles);
+                newGTs.add(gb.make());
+            }
+        }
+        vcBuilder.genotypes(newGTs);
+        return vcBuilder.make();
+    }
+
+    /**
+     * Create the new GenotypesContext with the subsetted PLs and ADs
+     * <p>
+     * Will reorder subsetted alleles according to the ordering provided by the list allelesToKeep
+     *
+     * @param originalGs      the original GenotypesContext
+     * @param originalAlleles the original alleles
+     * @param allelesToKeep   the subset of alleles to use with the new Genotypes
+     * @return a new non-null GenotypesContext
+     */
+    public static GenotypesContext subsetAlleles(final GenotypesContext originalGs,
```

Comment:
It's probably good if someone else takes a look at this.  It looks sane to me but I'm always confused by gls vs pls.